pkgs/development/tools/vcpkg/default.nix

```diff
@@ -0,0 +1,47 @@
+{ fetchFromGitHub
+, stdenvNoCC
+, lib
+, vcpkg-tool
+, writeShellScript
+}:
+
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = "vcpkg";
+  version = "2023.08.09";
+
+  src = fetchFromGitHub {
+    owner = "microsoft";
+    repo = "vcpkg";
+    rev = finalAttrs.version;
+    hash = "sha256-EK8Cikxn4yYvU6/xofNCUBEdJeGMNuoVRHHq+3sFYeA=";
+  };
+
+  installPhase = let
+    # vcpkg needs two directories to write to that is independent of installation directory.
+    # Since vcpkg already creates $HOME/.vcpkg/ we use that to create a root where vcpkg can write into.
+    vcpkgScript = writeShellScript "vcpkg" ''
+      vcpkg_writable_path="$HOME/.vcpkg/root/"
+
+      VCPKG_ROOT="@out@/share/vcpkg" ${vcpkg-tool}/bin/vcpkg \
+        --x-downloads-root=$vcpkg_writable_path/downloads \
+        --x-buildtrees-root=$vcpkg_writable_path/buildtrees \
+        --x-packages-root=$vcpkg_writable_path/packages \
+        "$@"
+      '';
+    in ''
```

Comment:
don't forget runHook

Reply:
Thanks for this one, I didn't know they had to be called explicitly. I don't see how to properly setup an install phase anywhere in the docs, the only thing I found in the manual is [this](https://nixos.org/manual/nixpkgs/stable/#ssec-install-phase), which tell very little information about how the install phase work and how to write one.

I found the relevant information here: https://github.com/jtojnar/nixpkgs-hammering/blob/main/explanations/missing-phase-hooks.md